kaggle/src/connector.js

```diff
@@ -0,0 +1,291 @@
+  var field = {};
+  field.name = 'c' + index;
+  field.label = columnName;
+  if (isNaN(content)) {
+    field.dataType = 'STRING';
+    field.semantics = {};
+    field.semantics.conceptType = 'DIMENSION';
+  } else {
+    field.dataType = 'NUMBER';
+    field.semantics = {};
+    field.semantics.conceptType = 'METRIC';
+    field.semantics.isReaggregatable = true
+  };
+  return field;
+};
+
+function processData(data, fields) {
+  var header = data[0];
+  var dataIndexes = fields.map( function(field){
+    return findIndex(field.label, header);
+  });
+  
+  var result = [];
+  for (var rowIndex = 1; rowIndex < data.length; rowIndex++) {
+    var rowData = dataIndexes.map( function(columnIndex){
+      return data[rowIndex][columnIndex];
+    }); 
+    result.push({
+      'values': rowData
+    });
+  }
+
+  return result;
+}
+
+function findIndex(value, array) {
+  return array.indexOf(value);
+}
+
+function getFileData(config) {
+  var userProperties = PropertiesService.getUserProperties();
+  var user = userProperties.getProperty('USERNAME');
+  var key = userProperties.getProperty('KEY');
+  var kaggleAuth = {
+    userName: user,
+    apiToken: key
+  };
+
+  var ownerSlug = config.ownerSlug;
+  var datasetSlug = config.datasetSlug;
+
+  var fileName = config.fileName;
+  var url = [
+    "datasets/download-raw",
+    ownerSlug,
+    datasetSlug,
+    fileName
+  ].join("/");
+
+  var key = [
+    ownerSlug,
+    datasetSlug,
+    fileName
+  ].join("--");
+
+  var fileContent = unZipCachedContent(key);
+
+  if ( fileContent === null) {
+    var response = kaggleFetch(url, kaggleAuth);
+    var fileContent = response.getContentText();
+    zipAndCacheContent(key, fileContent, cacheDuration);
+  }
+  
+  var csvData = Utilities.parseCsv(fileContent);
+  return csvData;
+}
+
+function kaggleFetch(url, kaggleAuth) {
+  var fullUrl = "https://www.kaggle.com/api/v1/" + url;
+  var authParamPlain = kaggleAuth.userName + ":" + kaggleAuth.apiToken;
+  var authParamBase64 = Utilities.base64Encode(authParamPlain);
+  var options = {
+    headers: {
+      "Authorization": "Basic " + authParamBase64 
+    }
+  };
+  var response = UrlFetchApp.fetch(fullUrl, options);
+  return response;
+}
+
+
+function compress(content) {
+  var textBlob = Utilities.newBlob(content);
+  var gzipBlob = Utilities.gzip(textBlob);
+  return gzipBlob;
+}
+
+function deCompress(blob) {
+  var uncompressedBlob = Utilities.ungzip(blob);
+  var content = uncompressedBlob.getDataAsString();
+  return content;
+}
+
+function zipAndCacheContent(key, content, duration) {
+  var gzipBlob = compress(content);
+  var cache = CacheService.getScriptCache();
+  cache.put(key, gzipBlob, duration);
+}
+
+
+function unZipCachedContent(key) {
+  var cache = CacheService.getScriptCache();
+  var gzipBlob = cache.get(key);
+  if (gzipBlob !== null) {
+    return deCompress(gzipBlob);
+  }
+  return null;
+}
+
+function isAdminUser() {
+  return true;
```

Comment:
This should either always return false, or run some logic to determine if the current user of the connector is an Admin.